src/pages/Dashboard.tsx

```diff
@@ -10,28 +10,49 @@ import {addFiledRequest, deleteFiledRequest, getFiledRequests} from "../actions/
 import {FiledRequest} from "@interfaces/filedRequest";
 import {DEMO_REQUESTS} from "../databases/filedRequests";
 import {DEMO_INSIGHTS} from "../databases/insights";
+import {addUserToGroup, getUser, removeUserFromGroup} from "../actions/userActions";
+import {TEST_USER, User} from "@interfaces/user";
 
 const Dashboard = () => {
+    const [currentViewedGroup, setCurrentViewedGroup] = useState<string>(TEST_GROUP);
+    const [currentUserGroups, setUserGroups] = useState<string[]>([TEST_GROUP]);
     const [displayedRequests, setDisplayedRequests] = useState<{[id: string]: FiledRequest}>(DEMO_REQUESTS);
     const [displayedInsights, setDisplayedInsights] = useState<{[id: string]: Insight}>(DEMO_INSIGHTS);
+    const [newGroupNameInput, setNewGroupNameInput] = useState<string>("");
 
     useEffect(() => {
         (async () => {
             try {
-                const storedRequests = await getFiledRequests(TEST_ORGANIZATION, TEST_GROUP);
-                setDisplayedRequests({...displayedRequests, ...storedRequests});
+                // TODO User should be fetched and persisted upon login
+                const user: User = await getUser(TEST_USER);
+                setUserGroups([TEST_GROUP].concat(user.groupsIds));
```

Comment:
state updating is not guaranteed to be a synchronous action in react, I would move the loadRequestsAnInsights in a second useEffect with currentUserGroups in the dependency array.